aws/lambda/outbound-proxy-vpc/main.tf

```diff
@@ -0,0 +1,89 @@
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/25" # 128 IPs
+
+  tags = var.tags
+}
+
+resource "aws_subnet" "public" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.0.0/26" # 64 IPs
+
+  tags = merge(var.tags, { Name = "${var.name}-public" })
+}
+
+resource "aws_subnet" "private" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.0.64/26" # 64 IPs
+
+  tags = merge(var.tags, { Name = "${var.name}-private" })
+}
+
+resource "aws_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+
+  tags = var.tags
+}
+
+resource "aws_nat_gateway" "main" {
+  allocation_id = var.eip_allocation_id
```

Comment:
I think with a few changes, this could have more use cases than as a outbound proxy. If a public ip becomes optional (and a default one would then be created), this could also be used as a base for a ec2 module with load balancer and asg. As far as I can tell there is nothing specific to the lambda, we could basically put anything inside the private subnet

Maybe we can/should rename this module to something like `private-subnet-with-internet-access` (maybe something a bit catchier 😂 )